nautobot/extras/models/groups.py

```diff
@@ -17,16 +18,140 @@
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        related_name="static_groups",
+        limit_choices_to=StaticGroupModelsQuery(),
+        help_text="The type of object contained in this Static Group.",
+    )
+
+    clone_fields = ["content_type"]
+    is_static_group_associable_model = False
+
+    class Meta:
+        ordering = ["content_type", "name"]
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def model(self):
+        """
+        Access to the underlying Model class for this group's `content_type`.
+
+        This class object is cached on the instance after the first time it is accessed.
+        """
+        if getattr(self, "_model", None) is None:
+            try:
+                self._model = self.content_type.model_class()
+            except models.ObjectDoesNotExist:
+                self._model = None
+        return self._model
+
+    @property
+    def members(self):
+        """Return the member objects for this group."""
+        # Since associated_object is a GenericForeignKey, we can't just do:
+        #     return self.static_group_associations.values_list("associated_object", flat=True)
+        return self.model.objects.filter(
+            pk__in=self.static_group_associations.values_list("associated_object_id", flat=True)
+        )
+
+    def clean(self):
+        super().clean()
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+
+            if self.content_type != database_object.content_type:
+                raise ValidationError({"content_type": "ContentType cannot be changed once created"})
+
+    def _check_valid_member(self, member):
+        if not isinstance(member, self.model):
+            raise TypeError(f"{member} is not a {self.model._meta.label_lower}")
+
+    def add_members(self, objects_to_add):
+        """Add the given list or QuerySet of objects to this Static Group."""
+        for obj in objects_to_add:
+            self._check_valid_member(obj)
+
+        for obj in objects_to_add:
+            # We don't use `.bulk_create()` currently because we want change logging for these creates.
+            # Might be a good future performance improvement though.
+            StaticGroupAssociation.objects.get_or_create(
+                static_group=self, associated_object_type=self.content_type, associated_object_id=obj.pk
+            )
+
+    def remove_members(self, objects_to_remove):
+        """Remove the given list or QuerySet of objects from this Static Group."""
+        pks_to_remove = set()
+        for obj in objects_to_remove:
+            self._check_valid_member(obj)
+            pks_to_remove.add(obj.pk)
+
+        StaticGroupAssociation.objects.filter(
+            static_group=self, associated_object_type=self.content_type, associated_object_id__in=pks_to_remove
+        ).delete()
```

Comment:
If a queryset is passed in would it be better to use the queryset values instead of building our own list of pks? Django might be able to make a more optimized query?